app/helpers/template_helpers.py

```diff
@@ -42,6 +42,90 @@ def get_page_header_context(language, theme):
     return context.get(theme)
 
 
+def get_footer_context(language_code, theme, base_url, contact_us_url, sign_out_url):
+    footer_urls = get_footer_urls(language_code, theme)
+
+    footer_warning = (
+        lazy_gettext(
+            f"Make sure you <a href='{sign_out_url}'>leave this page</a> or close your browser if using a shared device"
+        )
+        if request.blueprint == "post_submission"
+        else None
+    )
+
+    default_context = {
+        "crest": True,
+        "newTabWarning": lazy_gettext("The following links open in a new tab"),
+        "copyrightDeclaration": {
+            "copyright": lazy_gettext(
+                "Crown copyright and database rights 2020 OS 100019153."
+            ),
+            "text": lazy_gettext("Use of address data is subject to the"),
+            "link": lazy_gettext("terms and conditions"),
+            "url": f"{base_url}/{footer_urls['terms_and_conditions_path']}",
+            "target": "_blank",
+        },
+        "footerWarning": footer_warning,
```

Comment:
The `footerWarning` property shouldn't be set if there is no warning. It would be better to update the context if applicable before returning from this method.